src/bytes_mut.rs

```diff
@@ -819,6 +819,79 @@ impl BytesMut {
         }
     }
 
+    /// Attempts to reclaim the whole allocation of the `BytesMut`.
+    ///
+    /// If the `BytesMut` is empty but the underlying storage has been used before,
+    /// it might be possible to cheaply reclaim capacity as long as there are no
+    /// other `Bytes` or `BytesMut` handles alive.
+    ///
+    /// Returns `true` if the allocation was reclaimed (or reclaiming was not
+    /// necessary, because the underlying storage was empty); returns `false` if the
+    /// `BytesMut` is not empty or there are other outstanding references to the
+    /// underlying storage.
```

Comment:
It's not super clear that `try_reclaim` always returns `false` if the `BytesMut` is non-empty. I would probably have a paragraph that says something along the lines of "this method assumes that the `BytesMut` is empty and always returns false otherwise".